test/NuGet.Core.Tests/NuGet.PackageManagement.Test/DependencyGraphSpecTests.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using NuGet.ProjectModel;
+using Xunit;
+
+namespace NuGet.Test
+{
+    public class DependencyGraphSpecTests
+    {
+        [Fact]
+        public void DependencyGraphSpecTests_WithReplacedPackageSpecs()
```

Comment:
This test is not replacing any specs. 

You start with specA, and then you just add Specs B & C to it. 
Not replacing occurs.

Reply:
It was existing naming pattern [here](https://github.com/NuGet/NuGet.Client/blob/508808ef8ed761179d4c214551e9505d005e5aac/src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs#L474).
I just followed it, now I renamed it to `WithPackageSpecs`.